src/NBitcoin/ConsensusOptions.cs

```diff
@@ -2,18 +2,26 @@
 {
     /// <summary>
     /// An extension to <see cref="Consensus"/> to enable additional options to the consensus data.
+    /// TODO: Make immutable.
     /// </summary>
     public class ConsensusOptions
     {
-        /// <summary>The maximum allowed size for a serialized block, in bytes (only for buffer size limits).</summary>
-        public int MaxBlockSerializedSize { get; set; }
+        /// <summary>
+        /// Flag used to detect SegWit transactions.  
+        /// </summary>
+        public const int SerializeTransactionNoWitness = 0x40000000;
```

Comment:
In a perfect world this will go in to a `SegWitConsensusOptions` 
Eventually we will get there ;)